core/src/sumeragi/main_loop.rs

```diff
@@ -410,14 +361,18 @@ fn replace_top_block(
         "Replacing top block"
     );
 
-    update_topology(state, &committed_block);
+    update_topology(state, sumeragi, &committed_block);
 
     sumeragi.kura.replace_top_block(committed_block);
 
     cache_transaction(state, sumeragi)
 }
 
-fn update_topology(state: &mut State, committed_block: &VersionedCommittedBlock) {
+fn update_topology(
+    state: &mut State,
+    sumeragi: &Sumeragi,
+    committed_block: &VersionedCommittedBlock,
+) {
```

Comment:
It can be, but before this i want to do some refactoring in terms of relations between ‘Sumeragi’ and ’State’ because there bound isn't really clear to me and we almost always use them together.

Reply:
Methods aren't always better. IMO just a `fn` is enough.

Reply:
@Arjentix can we resolve this?